### PR TITLE
Fixes #1329 - encrypt BMC password in DB

### PR DESCRIPTION
**Pull request by amirfefer**

### CODE REVIEW

File: app/models/concerns/enc_output.rb [hunk 1]
```diff
@@ -43,6 +43,7 @@ module EncOutput
   def to_enc
     own_attributes = {}
     self.attributes.each do |k, v|
+      v = send(k) if v.is_a? String and v.include? Encryptable::ENCRYPTION_PREFIX
```
Comment (amirfefer) [inline app/models/concerns/enc_output.rb]: Encrypted attributes should be decrypted while `to_enc` invoked.

Comment (tbrisker) [inline app/models/concerns/enc_output.rb]: `v.starts_with?` would be better I think.
Also, use `&&` not `and` (http://devblog.avdi.org/2010/08/02/using-and-and-or-in-ruby/).

**Comment by iNecas** [comment]
[test]

File: app/models/concerns/enc_output.rb [hunk 2]
```diff
@@ -43,6 +43,7 @@ module EncOutput
   def to_enc
     own_attributes = {}
     self.attributes.each do |k, v|
+      v = send(k) if (v.is_a?(String) && v.start_with?(Encryptable::ENCRYPTION_PREFIX))
```
Comment (iNecas) [inline app/models/concerns/enc_output.rb]: I would prefer using more high-level methods than regexp matching and then calling generic `send`. What about something like:

``` ruby
v = decrypt_field(v) if self.is_a?(Encyrptable) && is_decryptable?(v)
```

Comment (amirfefer) [inline app/models/concerns/enc_output.rb]: @iNecas thanks! much more readable.

**Comment by iNecas** [comment]
:+1: works for me

File: test/unit/nics/bmc_test.rb [hunk 3]
```diff
@@ -32,6 +32,25 @@ def setup
...
+      host = FactoryGirl.build(:host, :managed)
+      bmc_nic = FactoryGirl.build(:nic_bmc, :with_subnet, :host => host)
+      bmc_nic.expects(:encryption_key).at_least_once.returns('25d224dd383e92a7e0c82b8bf7c985e815f34cf5')
+      bmc_nic.expects(:validate_bmc_proxy).at_least_once.returns(true)
+      bmc_nic.save
+      assert bmc_nic.password_in_db.include? Encryptable::ENCRYPTION_PREFIX
+    end
+
+    test 'BMC password is decrypted in ENC' do
+      host = FactoryGirl.build(:host, :managed)
```
Comment (dLobatog) [inline test/unit/nics/bmc_test.rb]: Please DRY and wrote a `setup` block :smile:

**Comment by tbrisker** [comment]
Merged as 977f12ed2fc010f72bf6c18b74130886c9460ac4, thanks @amirfefer!

**Pull request closed without merging**